packages/remix-dev/cli/run.ts

```diff
@@ -344,10 +346,13 @@ export async function run(argv: string[] = process.argv.slice(2)) {
       await commands.setup(input[1]);
       break;
     case "migrate": {
-      let { migrationId, projectDir } = await commands.migrate.resolveInput({
-        migrationId: flags.migration,
-        projectDir: input[1],
-      });
+      let projectDir = commands.migrate.resolveProjectDir(input[1]);
+      if (!flags.dry) {
+        checkGitStatus(projectDir, { force: flags.force });
+      }
+      let migrationId = await commands.migrate.resolveMigrationId(
+        flags.migration
+      );
```

Comment:
Maybe move this back into a `resolveInput` function and do the check right there?

That way the CLI code stays clean:
1. resolve the input
2. run the command

Reply:
We want to show the git status check as soon as possible (i.e. before potentially asking the user for the migration they want to apply). That optimizes for the UX of the CLI.

Reply:
If we have

```ts
export const resolveInput = () => {
  let projectDir = resolveProjectDir(input[1]);

  if (!flags.dry) {
    checkGitStatus(projectDir, { force: flags.force });
  }

  let migrationId = await resolveMigrationId(flags.migration);

  return { migrationId, projectId };
}
```

we would still have it as early as we have now, but the CLI file is cleaner and everything stays in the `migrate` folder